File: python/sparkts/onsMethods/Estimation.py
from pyspark.sql import DataFrame


class Estimation():

    defaultCol = 'EstResult'

    def __init__(self, df = None):

        if df is None : raise TypeError

        self._df = df

        self._jEst = self._df._sc._jvm.com.ons.sml.api.java.methods.JavaEstimationFactory.estimation(self._df._jdf)

    def __mandatoryArgumentCheck(self, arg1):

        if arg1 is None : raise TypeError


    def est1(self, df = None, inputCol = None, factor = 1, outputCol = defaultCol):

        self.__mandatoryArgumentCheck(inputCol)

        if df is None: df = self._df

        return DataFrame(self._jEst.est1(df._jdf, inputCol, factor, outputCol),  df.sql_ctx)

    def est2(self, df = None, inputCol = None, factor = 1, outputCol = defaultCol):

        self.__mandatoryArgumentCheck(inputCol)

        if df is None: df = self._df

        if factor == 0 : raise Exception('Factor cannot be zero')

        return DataFrame(self._jEst.est2(df._jdf, inputCol, factor, outputCol),  df.sql_ctx)




def estimation(df) :

    return Estimation(df)

File: src/main/scala/com/ons/sml/businessMethods/methods/Estimation.scala

package com.ons.sml.businessMethods.methods

import com.ons.sml.businessMethods.impl.EstimationImpl
import org.apache.spark.sql.DataFrame


class Estimation (val dfIn: DataFrame){

  if (dfIn == null ) throw new Exception("DataFrame cannot be null")

  val defaultCol = "EstResult"

  import EstimationImpl._

  private def mandatoryArgCheck(arg1 : String) : Unit = {

    if (arg1 == null) throw new Exception("Missing mandatory argument")
  }

  def est1(df: DataFrame, inputCol : String, factor : Integer, outputCol : String = defaultCol)  : DataFrame = {

    mandatoryArgCheck(inputCol)

    val dF = if (df == null) dfIn else df

    dF.checkColNames(Seq(inputCol))
      .est1(inputCol, factor, outputCol)
  }

  def est2(df: DataFrame, inputCol : String, factor : Integer, outputCol : String = defaultCol)  : DataFrame = {

    mandatoryArgCheck(inputCol)

    val dF = if (df == null) dfIn else df

    dF.checkColNames(Seq(inputCol))
      .est2(inputCol, factor, outputCol)
  }

}

object Estimation {

  def estimation(df: DataFrame) : Estimation = {new Estimation(df)}

}
